### PR TITLE
SourceDefinitions to DataSourceBindings

### PR DESCRIPTION
**Pull request by romil07**
Updating the vstest task to define dataSourceBindings instead of sourceDefinitions as per the new pattern

This is needed to support pagination in the test plan data source

### CODE REVIEW

**Comment by ShreyasRmsft** [comment]
@romil07 can you please list all the validations you have done?
Have you kicked off a few test plan - test suite runs to ensure values are being passed correctly?

File: Tasks/VsTest/task.json [hunk 1]
```diff
@@ -462,26 +462,29 @@
...
         {
             "target": "testPlan",
-            "endpoint": "/$(system.teamProject)/_apis/test/plans?filterActivePlans=true&api-version=3.0-preview.2",
-            "selector": "jsonpath:$.value[*].name",
-            "keySelector": "jsonpath:$.value[*].id",
-            "authKey": "tfs:teamfoundation"
+            "endpointId": "tfs:teamfoundation",
+            "endpointUrl": "{{endpoint.url}}/{{system.teamProject}}/_apis/test/plans?filterActivePlans=true&api-version=3.0-preview.2&$skip={{skip}}&$top=1000",
+            "resultSelector": "jsonpath:$.value[*]",
+            "resultTemplate": "{ \"Value\" : \"{{{id}}}\", \"DisplayValue\" : \"{{{name}}}\" }",
```
Comment (nigurr) [inline Tasks/VsTest/task.json]: 100 by default should be good enough. Can you please share the new UI experience with this change?

File: Tasks/VsTest/task.json [hunk 2]
```diff
@@ -462,26 +462,29 @@
...
         {
             "target": "testConfiguration",
-            "endpoint": "/$(system.teamProject)/_apis/test/configurations?api-version=3.0-preview.1",
-            "selector": "jsonpath:$.value[*].name",
-            "keySelector": "jsonpath:$.value[*].id",
-            "authKey": "tfs:teamfoundation"
+            "endpointId": "tfs:teamfoundation",
+            "endpointUrl": "{{endpoint.url}}/{{system.teamProject}}/_apis/test/configurations?api-version=3.0-preview.1",
+            "resultSelector": "jsonpath:$.value[*]",
+            "resultTemplate": "{ \"Value\" : \"{{{id}}}\", \"DisplayValue\" : \"{{{name}}}\" }"
```
Comment (nigurr) [inline Tasks/VsTest/task.json]: small change. can we display "{{{Id}}} - {{{name}}}"

File: Tasks/VsTest/task.json [hunk 3]
```diff
@@ -462,26 +462,29 @@
...
         },
         {
             "target": "testSuite",
-            "endpoint": "/$(system.teamProject)/_apis/test/plans/$(testPlan)/suites?$asTreeView=true&api-version=3.0-preview.2",
-            "selector": "jsonpath:$.value[*]",
-            "authKey": "tfs:teamfoundation"
+            "endpointId": "tfs:teamfoundation",
+            "endpointUrl": "{{endpoint.url}}/{{system.teamProject}}/_apis/test/plans/{{testPlan}}/suites?$asTreeView=true&api-version=3.0-preview.2",
```
Comment (nigurr) [inline Tasks/VsTest/task.json]: We may no longer require 'asTreeView' and this api supports pagination. Can we check this once?

File: Tasks/VsTest/task.json [hunk 4]
```diff
@@ -462,26 +462,29 @@
...
             "target": "testPlan",
-            "endpoint": "/$(system.teamProject)/_apis/test/plans?filterActivePlans=true&api-version=3.0-preview.2",
-            "selector": "jsonpath:$.value[*].name",
-            "keySelector": "jsonpath:$.value[*].id",
-            "authKey": "tfs:teamfoundation"
+            "endpointId": "tfs:teamfoundation",
+            "endpointUrl": "{{endpoint.url}}/{{system.teamProject}}/_apis/test/plans?filterActivePlans=true&api-version=3.0-preview.2&$skip={{skip}}&$top=1000",
+            "resultSelector": "jsonpath:$.value[*]",
+            "resultTemplate": "{ \"Value\" : \"{{{id}}}\", \"DisplayValue\" : \"{{{name}}}\" }",
+            "callbackContextTemplate": "{\"skip\": \"{{add skip 1000}}\"}",
```
Comment (GitHubSriramB) [inline Tasks/VsTest/task.json]: don't we need #add indicating it is a mustache helper?

File: Tasks/VsTest/task.json [hunk 5]
```diff
@@ -462,26 +462,29 @@
...
+            "callbackRequiredTemplate": "{{compareValueWithTop result.resultSize 1000}}",
```
Comment (GitHubSriramB) [inline Tasks/VsTest/task.json]: keep the function name generic - compareInt?

Comment (GitHubSriramB) [inline Tasks/VsTest/task.json]: don't we need #compareInt to indicate it is a mustache function?

File: Tasks/VsTest/task.json (hunk 4 above)
Comment (romil07) [reply Tasks/VsTest/task.json]: As discussed, we don't.

File: Tasks/VsTest/task.json (hunk 1 above)
Comment (romil07) [reply Tasks/VsTest/task.json]: Are you sure 100 is enough? Because we are not stopping the loop. If there are many items, there will be too many calls.

Comment (romil07) [reply Tasks/VsTest/task.json]: Also, from the experience side, there are no changes in UI. Just that user will feel a little delay while the results are fetched.

File: Tasks/VsTest/task.json (hunk 3 above)
Comment (nigurr) [reply Tasks/VsTest/task.json]: ignore this.

Comment (romil07) [reply Tasks/VsTest/task.json]: As discussed, ignoring this.

File: Tasks/VsTest/task.json (hunk 5 above)
Comment (romil07) [reply Tasks/VsTest/task.json]: Done

Comment (romil07) [reply Tasks/VsTest/task.json]: This was working fine.

File: Tasks/DownloadBuildArtifacts/task.json [hunk 6]
```diff
@@ -148,12 +148,15 @@
     {
       "endpointId": "tfs:teamfoundation",
       "target": "definition",
-      "endpointUrl": "{{endpoint.url}}/{{project}}/_apis/build/definitions?$top=1000",
+      "endpointUrl": "{{endpoint.url}}/{{project}}/_apis/build/definitions?api-version=3.0-preview&$top=1000&continuationToken={{{continuationToken}}}&queryOrder=2",
```
Comment (nigurr) [inline Tasks/DownloadBuildArtifacts/task.json]: why are we adding api-version explicitly? I believe latest version of API is much higher

**Pull request closed without merging**